### ISSUE TITLE
features needed for re-miniAOD of 2016 data/MC in 94X

### ISSUE THREAD

**Issue by slava77** [cms-sw/cmssw#22978]
This issue is to collect the list of PRs needed for re-miniAOD of 2016 data/MC (using inputs from 80X "legacy" AOD). 
- The release target for production is 94X.  PRs should be submitted to the master and CMSSW_9_4_MAOD_X branches
    - Any new feature should be integrated in the master branch first.
- Pure additions of products can be made in the generic part of the job configuration. 
    - The expectation is that the miniAOD for 2016 should be as consistent as possible with the 2017 data/MC re-miniAOD already made in 9_4_5/6 release
- All specific changes should appear either as a part of run2_miniAOD_80XLegacy (preferred) or as a customization function and should be enabled for the standard matrix tests.

[updated May 25]
- [x] #22713 Fix hot HCAL channels in legacy 2016 (merged May 25)
- [x] #23169 Adapting E/gamma miniAOD for the 80X re-miniAOD V2 (merged May 15)
- [x] #23044 GT: Change HcalRespCorrs and updated BS (merged Apr 25)
- [x] #23255  GTs 2016,2017 JES and JER (merged May 22)

Release target: ASAP

**Comment by slava77** [comment]
assign reconstruction,alca,pdmv

**Comment by slava77** [comment]
assign analysis

**Comment by prebello** [comment]
please test

**Comment by arunhep** [comment]
@prebello this is a github issue and not a PR. "please test" does not make a sense here.

**Comment by deguio** [comment]
hi @slava77 ,
a new GT will be needed as well. following up with @arunhep at:
https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/3495/1/1/1/1/1.html

**Comment by slava77** [comment]
@fabozzi @prebello @GurpreetSinghChahal

A related discussion has opened in
https://github.com/cms-sw/cmssw/pull/22456#discussion_r181999569

>>    I see. So, there are no relvals which run only miniAOD with DQM included?

> we do not produce relval 136.7611 (for instance) when we have a validation campaign. We could add this into the regular set (it will be needed anyway to test the fix)

>>    How are you planning to test the reminiAOD of 2016?

> Same as for reminiAOD of 2017: there should be a dedicated relval campaign

I just wanted to confirm/clarify that we have a full working relval setup to test this re-miniAOD.
For the PR tests I'm assuming that the setups used in 136.7611 for data and 1325.5 for MC will suffice.
Please confirm.

**Comment by arizzi** [comment]
before we cancel tomorrow's XPOG meeting for lack of requests, is there anything pending for legacy reminiaod (from this issue and related PRs) that you want to discuss in the meeting?

**Comment by slava77** [comment]
@lpernie @arunhep 
do we expect another PR with GT updates? 
Some JEC updates were mentioned during the ORP on Tuesday.

**Comment by fabiocos** [comment]
As the master version #22941 of #22713 has been merged, is there anything keeping the signature on the 9_4_MAOD_X version on hold?

**Comment by fabiocos** [comment]
@slava77 with the merge of #22713 for test the program discussed here seems to be completed, am I correct? The further needed ingredient is the JEC, that are anyway a GT issue. @arunhep @lpernie do you have any news about this?

**Comment by lpernie** [comment]
Conditions are completed. Sangeun confirmed that with our last PR no other conditions need to be added for jets.

**Comment by slava77** [comment]
I have updated the PR description with the check-marked PRs.
It looks like we are ready to go.

**Comment by fabiocos** [comment]
@slava77 @lpernie thank you, as far as I can see al the components needed for a test are ready

**Comment by fabiocos** [comment]
@slava77 @perrotta @previsualconsent just one question: is the external update https://github.com/cms-sw/cmsdist/pull/4036 corresponding to https://github.com/cms-data/EgammaAnalysis-ElectronTools/pull/4 also needed for this reprocessing?

**Comment by slava77** [comment]
On 5/28/18 4:26 AM, Fabio Cossutti wrote:
> @slava77 <https://github.com/slava77> @perrotta
> <https://github.com/perrotta> @previsualconsent
> <https://github.com/previsualconsent> just one question: is the external
> update cms-sw/cmsdist#4036 <https://github.com/cms-sw/cmsdist/pull/4036>
> corresponding to cms-data/EgammaAnalysis-ElectronTools#4
> <https://github.com/cms-data/EgammaAnalysis-ElectronTools/pull/4> also
> needed for this reprocessing?
> 


we agreed not to backport

**Comment by fabiocos** [comment]
@slava77 thank you, so we are ready for a test build. For my understanding: for which reason this change is not included? Are the changes irrelevant for the reprocessing?

**Comment by slava77** [comment]
On 5/28/18 5:36 AM, Fabio Cossutti wrote:
> @slava77 <https://github.com/slava77> thank you, so we are ready for a
> test build. For my understanding: for which reason this change is not
> included? Are the changes irrelevant for the reprocessing?
> 

the changes in the external are relevant both for 2017 and 2016.
The impact of the fix is rather small.
If we are going to rerun the 2017 miniAOD, then sure we can add the
external update as well.

**Comment by slava77** [comment]
> the changes in the external are relevant both for 2017 and 2016.

after more careful reading of discussion with @Sam-Harper , the update in the egamma smearings appears to be only for the 2016 case.
So, we can benefit from an update to the external (see more in the discussion in https://github.com/cms-data/EgammaAnalysis-ElectronTools/pull/4)

**Comment by slava77** [comment]
The bugfix in the electron momentum was signed off just now https://github.com/cms-sw/cmssw/pull/23708.
AFAIK, this is the only fix remaining in the miniAOD step in addition to 9_4_9_cand1

**Comment by fabiocos** [comment]
@slava77 I have merged it for test, we need to understand how to proceed for 9_4_9

**Comment by slava77** [comment]
closing this as done/resolved.
